python-frontend/src/main/java/org/sonar/python/semantic/v2/types/ProgramStateTypeInferenceVisitor.java

```diff
@@ -66,7 +65,7 @@ public void visitQualifiedExpression(QualifiedExpression qualifiedExpression) {
     }
   }
 
-  private static PythonType union(Stream<PythonType> types) {
-    return types.reduce(UnionType::or).orElse(PythonType.UNKNOWN);
+  private static PythonType union(Set<PythonType> types) {
```

Comment:
To explain my reasoning: 
- Passing streams as an argument feels slightly sketching since a stream can only be consumed/called once
- It simplifies the caller slightly without impacting this function